syntax = "proto3";

option java_multiple_files = true;
option java_package = "com.ni.grpc.devicerestricted";
option java_outer_classname = "NiDeviceRestricted";
option csharp_namespace = "NationalInstruments.Grpc.DeviceRestricted";

package nidevice_restricted_grpc;

service DebugSessionPropertiesRestricted {
  rpc IsDebugSessionEnabled(IsDebugSessionEnabledRequest) returns (IsDebugSessionEnabledResponse);
  rpc IsDebugSessionServerOutOfProcess(IsDebugSessionServerOutOfProcessRequest) returns (IsDebugSessionServerOutOfProcessResponse);
  rpc IsDebugSessionSupported(IsDebugSessionSupportedRequest) returns (IsDebugSessionSupportedResponse);
  rpc SetDebugSessionEnabled(SetDebugSessionEnabledRequest) returns (SetDebugSessionEnabledResponse);
  rpc SetDebugSessionServerOutOfProcess(SetDebugSessionServerOutOfProcessRequest) returns (SetDebugSessionServerOutOfProcessResponse);
}

message IsDebugSessionEnabledRequest {
  uint32 product_id = 1;
  string serial_number = 2;
}

message IsDebugSessionEnabledResponse {
  bool enabled = 1;
}

message IsDebugSessionServerOutOfProcessRequest {
  uint32 product_id = 1;
  string serial_number = 2;
}
 
message IsDebugSessionServerOutOfProcessResponse {
  bool out_of_process = 1;
}

message IsDebugSessionSupportedRequest {
  uint32 product_id = 1;
  string serial_number = 2;
}
 
message IsDebugSessionSupportedResponse {
  bool supported = 1;
}

message SetDebugSessionEnabledRequest {
  uint32 product_id = 1;
  string serial_number = 2;
  bool enabled = 3;
}

message SetDebugSessionEnabledResponse {}

message SetDebugSessionServerOutOfProcessRequest {
  uint32 product_id = 1;
  string serial_number = 2;
  bool out_of_process = 3;
}

message SetDebugSessionServerOutOfProcessResponse {}
